h/static/scripts/annotator/anchoring/types.coffee

```diff
@@ -47,7 +49,62 @@ class RangeAnchor
       endOffset: range.endOffset
     }
 
+###*
+# Converts between TextPositionSelector selectors and Range objects.
+###
+class TextPositionAnchor
```

Comment:
`html.coffee` assumes that Range <-> selector conversion for all 4 types of selector has the same interface, which looks something like this:

```ts
interface AnchorFactory {
  fromRange(root: Element, range: Range, options: Object): Anchor
  fromSelector(root: Element, selector: Selector): Anchor
}

interface Anchor {
  toSelector(): Selector;
  toRange(options: Object): Range;
}

interface Selector {
  type: 'TextQuoteSelector' | 'TextPositionSelector' | 'FragmentSelector' | 'RangeSelector';
  // Selector-specific fields...
}
```

The `TextPositionAnchor` and `TextQuoteAnchor` classes added here are wrappers around the interface exported by the latest version of the quote and position anchoring libraries so that this remains the case.

The other way to handle this would be to adopt the new interface of the `dom-anchor-text-{quote, position}` libraries as the interface for all 4 types of selector. I think that will be cleaner overall but I would like to explore that as a follow-up refactoring.